src/twisted/newsfragments/8375.feature

```diff
@@ -0,0 +1 @@
+twisted.application.internet.ClientService now accepts a function which is used to initialize a new connection as the onNewConnection argument.
```

Comment:
`is used to initialize a new connection` ... I don't know if I get what it takes to initialize a new connection?

By reading only this I can understand that the function is in charge of creating/making that connection.

Maybe:

twisted.application.internet.ClientService now has the onNewConnection argument which accepts a function called when a connection is made.